src/client/mod.rs

```diff
@@ -825,15 +850,66 @@ impl KafkaClient {
                                                                           now,
                                                                           req));
             for tp in resp.topic_partitions {
-                let e = res.entry(tp.topic).or_insert(vec![]);
-                for p in tp.partitions {
-                    e.push(p.into_offset());
+                let mut entry = res.entry(tp.topic);
+                let mut new_resp_offsets = None;
+                let mut err = None;
+
+                // Use an explicit scope here to allow insertion into a vacant entry
+                // below
+                {
+                    // Return a &mut to the response we will be collecting into to
+                    // return from this function. If there are some responses we have
+                    // already prepared, keep collecting into that; otherwise, make a
+                    // new collection to return.
+                    let resp_offsets = match entry {
+                        hash_map::Entry::Occupied(ref mut e) => e.get_mut(),
+                        hash_map::Entry::Vacant(_) => {
+                            new_resp_offsets = Some(Vec::new());
+
+                            match new_resp_offsets {
+                                Some(ref mut v) => v,
+                                None => panic!("Option value was None after Some assignment"),
+                            }
+                        }
+                    };
+
+                    for p in tp.partitions {
+                        let partition_offset = match p.into_offset() {
+                            Ok(po) => po,
+                            Err(code) => {
+                                err = Some((p.partition, code));
+                                break;
+                            }
+                        };
+
+                        resp_offsets.push(partition_offset);
+                    }
+                }
+
+                if let Some((partition, code)) = err {
+                    let topic = KafkaClient::get_key_from_entry(entry);
+                    return Err(Error::TopicPartitionError(topic, partition, code));
+                }
+
+                if let hash_map::Entry::Vacant(e) = entry {
+                    // unwrap is ok because if it is Vacant, it would have
+                    // been made into a Some above
+                    e.insert(new_resp_offsets.unwrap());
                 }
             }
```

Comment:
may i ask you to remove the empty lines within this `for tp in resp.topic_partitions {` loop?